cmd/fyne-cross/main.go

```diff
@@ -79,6 +81,7 @@ func main() {
 	flag.Usage = printUsage
 
 	defaultTarget := strings.Join([]string{build.Default.GOOS, build.Default.GOARCH}, "/")
+	flag.StringVar(&dockerImage, "docker-image", builder.DefaultDockerImage, "Docker image to use. Can be an image name or the path to a Dockerfile")
```

Comment:
we should figure out a way to specify a custom image for each target.
fyne-cross can accept multiple targets and in the feature we could split base images for each os/arch (see #25)

Reply:
I honestly see no need for having a per-target docker image, it would increase maintenance burden a lot. It would also require users to have a per target custom Dockerfile when they need to make changes. Invoking `fyne-cross` once per target seems a perfectly fine solution for me. Most of the time you will need to call target specific tools anyway (e.g. code signing for macOS and Windows).

Reply:
> I honestly see no need for having a per-target docker image, it would increase maintenance burden a lot.  It would also require users to have a per target custom Dockerfile when they need to make changes.

I agree that having a per-target image increase the maintenance burden but there are some other things we have to consider too. The fyne-cross objective is to support as much as possible all targets supported by Fyne. With that in mind along with the "desktop" image there is already a dedicated image for android we have to support. Additionally there are plans to support FreeBSD and iOS (although I don't know if this is even feasible right now :) ) that will have dedicated image too.

Least but not last #25 could allow to shrink image size and make it more usable even in zones when the internet connection speed is not high. Thinking of use cases where for example you are on darwin and want build only for another platform.

> Invoking fyne-cross once per target seems a perfectly fine solution for me. Most of the time you will need to call target specific tools anyway (e.g. code signing for macOS and Windows).

Fair enough, but in this case I'd prefer to slightly change the command line args/options (introducing a break change) and introduce something like sub commands to have it less error prone, like:
`fyne-cross <target> --arch=[] --docker-image="lucor/custom-target-image:latest"`

What do you think?

Reply:
> Fair enough, but in this case I'd prefer to slightly change the command line args/options (introducing a break change) and introduce something like sub commands to have it less error prone, like:
> `fyne-cross <target> --arch=[] --docker-image="lucor/custom-target-image:latest"`
> 
> What do you think?

I'm not sure how that's different than the `-docker-image="lucor/custom-target-image:latest"` flag implemented in this PR (except for the double dash). Could you please elaborate?

Reply:
The main point we need to address is how to have the possibility to specify a custom image per target (theoretically we'd allow also per arch but better leave this out now to avoid not needed over-engineering).

To achieve this some ways come in mind right now:
a) specify the custom image using a per os flag: `-docker-image-linux`, `-docker-image-windows`, -docker-image-android`, ecc..
b) specify the custom image using an env variable: `FYNE_CROSS_IMAGE_LINUX`, `FYNE_CROSS_ANDROID`, ecc...
c) update fyne-cross to use subcommands: 
```
fyne-cross <OS> --docker-image="<custom_image>" #builds <OS>
fyne-cross linux --docker-image="<custom_image>" #builds linux
fyne-cross android --docker-image="<custom_image>" #builds android
```

Solution c) would introduce a breaking change but in the long term could be a good choice since every OS could require specific options.

With regard to the `-docker-image="lucor/custom-target-image:latest"` what I meant was related to the value of the image could be only defined as:
- `<repository>:<tag>` to use remote or local tagged images (i.e. fiam/fyne-cross-windows:latest)
- the image id of a local image: (i.e `5d70d40d6b7c`)